### PR TITLE
EOS-21062: ConfStore: Support "merge" option in conf CLI

### PR DESCRIPTION
**Pull request by vastradparayya**
Signed-off-by: Parayya Vastrad <parayya.vastrad@seagate.com>

# Problem Statement
- ConfStore: Support "merge" option in conf CLI

# Design
-  [EOS-21062](https://jts.seagate.com/browse/EOS-21062)

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [x] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 1]
```diff
@@ -92,6 +92,28 @@ def diff(args) -> str:
                 stderr.decode('utf-8')
         return output
 
+    @staticmethod
+    def merge(args):
+        """ merges source and dest. conf files """
+
+        dest_index = ConfCli._index
+        ConfCli._index = 'src_index'
```
Comment (ujjwalpl) [inline py-utils/src/utils/conf_store/conf_cli.py:100]: Please avoid changing _index. Add a new interface for handling this. 
```
def _load(index, url): 
    Conf.load(index, url)
```

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/conf_store/conf_cli.py (hunk 1 above)
Comment (vastradparayya) [reply py-utils/src/utils/conf_store/conf_cli.py]: Done

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 2]
```diff
@@ -92,6 +97,25 @@ def diff(args) -> str:
                 stderr.decode('utf-8')
         return output
 
+    @staticmethod
+    def merge(args):
+        """ merges source and dest. conf files """
+
+        dest_index = ConfCli._index
+        ConfCli.load(args.src_url, 'src_index')
+        src_index = 'src_index'
```
Comment (ujjwalpl) [inline py-utils/src/utils/conf_store/conf_cli.py:106]: src_index should be declared earlier, i.e.. 
```
src_index = 'src_index'
dest_index = ConfCli._index
ConfCli.load(args.src_url, src_index)
....
```

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/conf_store/conf_cli.py (hunk 2 above)
Comment (vastradparayya) [reply py-utils/src/utils/conf_store/conf_cli.py:104]: Done

**Review by ujjwalpl — APPROVED** [review]

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 3]
```diff
@@ -198,6 +222,26 @@ def add_args(sub_parser) -> None:
...
             "e.g. In case of 'xxx[0],xxx[1]', only 'xxx' is returned\n\n")
 
+
+class MergeCmd:
+    """ Get Merge Cmd Structure """
+
+    @staticmethod
+    def add_args(sub_parser) -> None:
+        s_parser = sub_parser.add_parser('merge', help=
+            "Merges contents of source and destination conf files\n."
```
Comment (sachinpunadikar) [inline py-utils/src/utils/conf_store/conf_cli.py:232]: Merges contents of source file into destination conf files\n.

Comment (vastradparayya) [reply py-utils/src/utils/conf_store/conf_cli.py]: Done

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 4]
```diff
@@ -92,6 +97,25 @@ def diff(args) -> str:
                 stderr.decode('utf-8')
         return output
 
+    @staticmethod
+    def merge(args):
+        """ merges source and dest. conf files """
```
[the remark this answers is not in the document]
Comment (vastradparayya) [reply py-utils/src/utils/conf_store/conf_cli.py]: Done

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by selvakumaar5496 — APPROVED** [review]

**Comment by sachinpunadikar** [comment]
Deployment is successful 
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/PyUtils/502/

**Pull request merged**